controllers/remove_process_groups_test.go

```diff
@@ -489,17 +501,70 @@ var _ = Describe("remove_process_groups", func() {
+					Expect(len(cluster.Status.ProcessGroups)).To(Equal(15))
+				})
+			})
+
+			When("including a process which is excluded both by IP and locality", func() {
+				var removedProcessGroup *fdbv1beta2.ProcessGroupStatus
+
+				BeforeEach(func() {
+					removedProcessGroup = cluster.Status.ProcessGroups[0]
+					Expect(removedProcessGroup.ProcessGroupID).To(Equal(fdbv1beta2.ProcessGroupID("storage-1")))
+					removedProcessGroup.MarkForRemoval()
+					cluster.Status.ProcessGroups[0] = removedProcessGroup
+
+					status.Cluster.DatabaseConfiguration.ExcludedServers = append(status.Cluster.DatabaseConfiguration.ExcludedServers, fdbv1beta2.ExcludedServers{Locality: removedProcessGroup.GetExclusionString()})
+					status.Cluster.DatabaseConfiguration.ExcludedServers = append(status.Cluster.DatabaseConfiguration.ExcludedServers, fdbv1beta2.ExcludedServers{Locality: removedProcessGroup.Addresses[0]})
 
 					removedProcessGroups[removedProcessGroup.ProcessGroupID] = true
 				})
 
 				It("should include one process", func() {
-					fdbProcessesToInclude := getProcessesToInclude(cluster, removedProcessGroups)
+					fdbProcessesToInclude, err := getProcessesToInclude(logr.Logger{}, cluster, removedProcessGroups, status)
+					Expect(err).NotTo(HaveOccurred())
 					Expect(len(fdbProcessesToInclude)).To(Equal(2))
 					Expect(fdbv1beta2.ProcessAddressesString(fdbProcessesToInclude, " ")).To(Equal(fmt.Sprintf("%s %s", removedProcessGroup.GetExclusionString(), removedProcessGroup.Addresses[0])))
 					Expect(len(cluster.Status.ProcessGroups)).To(Equal(15))
 				})
 			})
+
+			When("one excluded process is missing from excluded servers", func() {
+				var removedProcessGroup *fdbv1beta2.ProcessGroupStatus
+				var removedProcessGroup2 *fdbv1beta2.ProcessGroupStatus
+
+				BeforeEach(func() {
+					removedProcessGroup = cluster.Status.ProcessGroups[0]
+					Expect(removedProcessGroup.ProcessGroupID).To(Equal(fdbv1beta2.ProcessGroupID("storage-1")))
+					removedProcessGroup.MarkForRemoval()
+					cluster.Status.ProcessGroups[0] = removedProcessGroup
+					removedProcessGroups[removedProcessGroup.ProcessGroupID] = true
+
+					removedProcessGroup2 = cluster.Status.ProcessGroups[1]
+					Expect(removedProcessGroup2.ProcessGroupID).To(Equal(fdbv1beta2.ProcessGroupID("storage-2")))
+					removedProcessGroup2.MarkForRemoval()
+					cluster.Status.ProcessGroups[1] = removedProcessGroup2
+					status.Cluster.DatabaseConfiguration.ExcludedServers = append(status.Cluster.DatabaseConfiguration.ExcludedServers, fdbv1beta2.ExcludedServers{Locality: removedProcessGroup2.GetExclusionString()})
+					removedProcessGroups[removedProcessGroup2.ProcessGroupID] = true
+				})
+
+				It("should include two process", func() {
+					fdbProcessesToInclude, err := getProcessesToInclude(logr.Logger{}, cluster, removedProcessGroups, status)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(len(fdbProcessesToInclude)).To(Equal(2))
```

Comment:
Since only `removedProcessGroup2` is present in the `ExcludedServer` with the locality based exclusion string, shouldn't `fdbProcessesToInclude` also include the IP address from `removedProcessGroup`?

Reply:
Since we are not using the locality based exclusion setting and we are using the excluded server list we are re including all the servers the way they are excluded. So for e.g if a process is excluded using locality it will be reincluded using locality and not by both. Similarly if a process is excluded using IP, it will be reincluded using IP.